### PR TITLE
[js/web] optimize bundle file size

### PR DESCRIPTION
**Pull request by fs-eire**
**Description**: This change applies multiple optimizations to reduce bundle file size:

1. use es2017 by default instead of es5 (add build target for es5 in case it's needed)
2. introduce a compile time build flag 'BUILD_DEFS' to instruct terser to drop unnecessary code. Add check to make sure those flags are not in code for production mode.
3. DISABLE_WASM_THREAD is defined but currently not in use yet.

file size:

608,510 ort.es5.min.js
545,529 ort.es6.min.js
538,793 ort.min.js
 79,922 ort.wasm-core.min.js
152,251 ort.wasm.min.js
394,524 ort.webgl.min.js

### CODE REVIEW

File: js/web/webpack.config.js [hunk 1]
```diff
@@ -80,13 +113,43 @@ function buildConfig({ filename, format, target, mode, devtool }) {
...
+          'Check BUILD_DEFS',
+          (compilation) => {
+            for (const filename of compilation.assetsInfo.keys()) {
+              if (filename.endsWith('.js')) {
+                const asset = compilation.assets[filename];
+                if (asset) {
+                  const content = asset.source();
+                  if (typeof content !== 'string') {
+                    throw new Error(`content for target file '${filename}' is not string.`);
+                  } if (content.includes('DISABLE_WEBGL')
```
Comment (hanbitmyths) [inline js/web/webpack.config.js:141]: new line

**Review by hanbitmyths — COMMENTED** [review]
can you add an instruction to js/README.md?

**Comment by fs-eire** [comment]
> can you add an instruction to js/README.md?

Added.

**Review by hanbitmyths — APPROVED** [review]
Great work for bundle size reduction!

**Pull request merged**